pandas/core/strings.py

```diff
@@ -1095,24 +1095,48 @@ def str_pad(arr, width, side='left', fillchar=' '):
 
 def str_split(arr, pat=None, n=None):
     """
+    Split strings around given separator/delimiter.
+
     Split each string (a la re.split) in the Series/Index by given
     pattern, propagating NA values. Equivalent to :meth:`str.split`.
 
     Parameters
     ----------
     pat : string, default None
-        String or regular expression to split on. If None, splits on whitespace
+        String or regular expression to split on. If None, split on whitespace.
     n : int, default -1 (all)
-        None, 0 and -1 will be interpreted as return all splits
+        * None, 0 and -1 will be interpreted as return all splits.
+        * Vary output dimensionality if `expand` is True:
+            - If n >= default splits, return all splits.
+            - If n < default splits, makes first n splits only.
     expand : bool, default False
```

Comment:
Looking at this in more detail, I see why you have this documented here in spite of the fact that `str_split` doesn't actually accept this parameter (it's docstring is copied to `split`). Ref some of the work in https://github.com/pandas-dev/pandas/pull/10085

@jreback is there any reason why we wouldn't want to deprecate items like `str_split` and move to private internal methods like `_str_split` that the front-end facing API uses instead? If so may be a logical follow up to this docstring update

Reply:
@WillAyd the `str_split` methods are not considered public, it's just how it is implemented right now. I agree this one is a bit confusing as some keyword are used in `str_split` and some only in the method, but generally in the file, the docstrings are located at the functions (so `str_split`), although they are written as if they were documenting the method.

Feel free to open an issue if you have a proposal how this could be improved.